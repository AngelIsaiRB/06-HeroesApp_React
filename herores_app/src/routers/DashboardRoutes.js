import React from 'react'
import { Navbar } from '../components/ui/NavBar'
import {Switch,Route, Redirect } from "react-router-dom";
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heores/HeroScreen';
import { DcScreen } from '../components/dc/DcScreen';

export const DashboardRoutes = () => {
    return (
        <>
         <Navbar/> 
         <div>
             <Switch>
                <Route  path="/marvel" component={MarvelScreen}/>
                <Route  path="/heroe/:heroeId" component={HeroScreen}/>
                <Route  path="/dc" component={DcScreen}/>
                <Redirect to="/marvel"/>

             </Switch>
        </div>  
        </>
    )
}
