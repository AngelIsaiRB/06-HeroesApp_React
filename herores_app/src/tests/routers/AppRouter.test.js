import React from 'react'
import { mount } from 'enzyme'
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('pruebas en el componente <AppRouter/>', () => {
    
    const contextValue ={
        dispatch: jest.fn(),
        user:{
            logged:false
        }
    }    
    test('should mostrar login si no estoy autentificado', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });

    test('should de mostrar el componente de marvel si esta autentificado', () => {
        const contextValue ={
            dispatch: jest.fn(),
            user:{
                logged:true,
                names:"isai",
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )
        expect(wrapper.find(".navbar").exists()).toBe(true);

    });
    
        
})
