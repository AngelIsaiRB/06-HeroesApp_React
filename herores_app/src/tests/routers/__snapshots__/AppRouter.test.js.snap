// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <AppRouter/> should mostrar login si no estoy autentificado 1`] = `
<div>
  <div
    clasname="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
